### ISSUE TITLE
[NVIDIA ONLY] Unoptimized driver code path leads to bad performance 

### ISSUE THREAD

**Issue by Yahfz** [RPCS3/rpcs3#9173]
From our discussion on discord, kd-11 mentioned that the performance with render target format R8G8B8A8_UNORM is much slower than using B8G8R8A8_UNORM for no obvious reason.
We do not want to use the latter since PS3 supports native A8G8B8R8 which maps cleanly onto the former with a simple 32-bit byteswap.

**Comment by Yahfz** [comment]
BGRA:
![BGRA](https://user-images.githubusercontent.com/17715080/97722544-85b5a000-1aa9-11eb-9266-75349139832f.png)
RGBA:
![RGBA](https://user-images.githubusercontent.com/17715080/97722550-86e6cd00-1aa9-11eb-9950-f98efbb1db09.png)

**Comment by SEt-t** [comment]
If the source format is actually A8B8G8R8 then it should map into B8G8R8A8_UNORM with rotation by 8 bits.
Is rotation here much slower than byteswap?

**Comment by kd-11** [comment]
> If the source format is actually A8B8G8R8 then it should map into B8G8R8A8_UNORM with rotation by 8 bits.
> Is rotation here much slower than byteswap?

Yes, very much so, but not for the reasons you're thinking of. There a few things you fail to take into account:
1. The operation goes both ways so yo need to implement both rotate right and rotate left. Byteswap is its own inverse transformation, simplifying this process.
2. The operation will almost never happen by itself and will have to merge with some other pass, usually a 16-bit byteswap. When doing byteswap_32 => byteswap_16, we have an optimized shader byteswap_32_16 that is also its own inverse transformation.

Adding rotate left and rotate right to the equation introduces more complex permutations and becomes annoying to maintain. The fact that 32-bit image copies are also not straightforward anymore, requiring compute shaders to maintain integrity, or god forbid, channel swap trickery which epically falls to bits when blit engine comes into play.

TL;DR - Rotates themselves are not slower than byteswaps, they are more of a maintenance burden and somewhat of a performance burden due to the loss of VkCmdCopyImage. RGBA8 completely solves this problem and allows all basic transformations to be handled in just 3 permutations of a compute task, opening up tons of optimizations.

**Comment by Yahfz** [comment]
Nvidia seems to have fixed the issue silently. 
Driver used: 547.51

**Comment by Yahfz** [comment]
after a more in-depth regression testing it seems like there's only one regression and its caused by #8713
